lib/logstash/filters/mutate.rb

```diff
@@ -285,7 +289,10 @@ def convert_string(value)
   end
 
   def convert_integer(value)
-    value.to_i
+    true.eql?(value) ? 1  #return 1 if a true boolean
```

Comment:
the nested ternary is pretty hard to follow; the following is equivalent, and requires fewer comments because the code is less opaque:

~~~ ruby
def convert_integer(value)
  return 1 if value == true
  return 0 if value == false

  return value if value.nil?
  return value if value.kind_of?(String) && value.empty?

  value.to_i
end
~~~

Reply:
....but I ❤️ ? :  ? : ? : ? 's 

Will update as suggested 😄